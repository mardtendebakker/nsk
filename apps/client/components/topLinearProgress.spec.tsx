import { render } from '@testing-library/react';
import { LinearProgress } from '@mui/material';
import TopLinearProgress, { hideProgress, showProgress } from './topLinearProgress';

jest.mock('@mui/material/LinearProgress', () => jest.fn(({ id }) => <div id={id} />));

describe('TopLinearProgress', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(<TopLinearProgress />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders with correct props', () => {
    render(<TopLinearProgress />);

    expect(LinearProgress).toBeCalledWith({
      sx: {
        position: 'fixed',
        display: 'none',
        left: 0,
        right: 0,
        top: 0,
        zIndex: 1202,
      },
      color: 'secondary',
      id: 'progress',
    }, {});
  });

  it('shows progress', () => {
    const { container } = render(<TopLinearProgress />);
    showProgress();
    const element = container.querySelector('#progress');

    expect(element.getAttribute('style')).toBe('display: unset;');
  });

  it('hides progress', () => {
    const { container } = render(<TopLinearProgress />);
    hideProgress();
    const element = container.querySelector('#progress');

    expect(element.getAttribute('style')).toBe('display: none;');
  });
});
