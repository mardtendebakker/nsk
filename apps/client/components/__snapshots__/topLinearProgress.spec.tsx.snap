// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopLinearProgress matches snapshot 1`] = `
<DocumentFragment>
  <div
    id="progress"
  />
</DocumentFragment>
`;
